src/ipyradiant/visualization/datashader_vis.py

```diff
@@ -0,0 +1,135 @@
+import traitlets as T
+
+import holoviews as hv
+import IPython
+import ipywidgets as W
+from bokeh.models import HoverTool
+from holoviews import streams
+from holoviews.operation.datashader import bundle_graph
+from rdflib import Graph
+from rdflib.extras.external_graph_libs import rdflib_to_networkx_graph
+
+from .base import NXBase
+
+
+class DatashaderVis(NXBase):
+    output = T.Instance(W.Output)
+    tooltip = T.Unicode(default_value="nodes")
+    tooltip_dict = T.Dict()
+    node_tooltips = T.List()
+    edge_tooltips = T.List()
+    sparql = T.Unicode()
+
+    @T.default("output")
+    def _make_default_output(self):
+        output = W.Output()
+        with output:
+            hv.extension("bokeh")
+        output.clear_output()
+        return output
+
+    @T.default("edge_tooltips")
+    def _make_edge_tooltip(self):
+        return [
+            ("Source", "@start"),
+            ("Target", "@end"),
+        ]
+
+    @T.default("node_tooltips")
+    def _make_node_tooltip(self):
+        return [
+            ("ID", "@index"),
+        ]
+
+    @T.default("tooltip")
+    def _make_tooltip(self):
+        return "nodes"
+
+    @T.default("tooltip_dict")
+    def _make_tooltip_dict(self):
+        return {
+            "nodes": HoverTool(tooltips=self.node_tooltips),
+            "edges": HoverTool(tooltips=self.edge_tooltips),
+        }
+
+    @T.default("sparql")
+    def _make_sparql(self):
+        return """
+            CONSTRUCT {
+                ?s ?p ?o .
+            }
+            WHERE {
+                ?s ?p ?o .
+                FILTER (!isLiteral(?o))
+                FILTER (!isLiteral(?s))
+            }
+            LIMIT 300
+        """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.children = [self.output]
+
+    def display_datashader_vis(self, p):
+        self.output.clear_output()
+        with self.output:
+            IPython.display.display(p)
+
+    def strip_and_produce_rdf_graph(self, rdf_graph: Graph):

```

Comment:
for less obvious methods we will want docstring too. This can be a single sentence that explains the basic reasoning and functionality.